### PR TITLE
Functionality Added for Tickets 5,6,8

### PR DESCRIPTION
**Pull request by ColemDev**
Adds functions for:

- ArticlesByTopic: Throughout the site there are links that link to the specific topic clicked. I hope to create a drop down nav menu to allow the user to use search and to navigate to all articles (so essentially all topics). In the mean time you can navigate to the list of all articles by adding /articles at the end of the url.
-  ArticlesById: Links that take the user to to the article to be read can be found underneath the summary of articles found in lists.
- CommentsByArticlesId will show a list of all comments related to the the article and can be found underneath the article in question on the article page (not on the articles list as this would make the user have to scroll a lot to get to the next article).
There is also a basic form on that page for the user to submit a comment though it is not yet usable and is only there as a place-holder to keep the basic shape of the application until comment editing functionality and styling has been implemented. 
- Home: the main header title is clickable and will take you to the landing page where you can click on the "most popular topics" 
- This will link you to the ArticlesByTopic list.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -20,8 +20,12 @@ const App = () => {
         <Routes>
           <Route path="/" element={<Home />} />
```
Comment (pkaodev) [inline src/App.js:21]: Lines 18+19 in App.js - If you're not using these components remove them from the code.

File: src/components/Home.jsx [hunk 2]
```diff
@@ -24,11 +26,22 @@ const Home = () => {
...
+            <Link to="/articles/topic/coding">Coding</Link>
+          </li>
+          <li>
+            <Link to="/articles/topic/cooking">Cooking</Link>
+          </li>
+          <li>
+            <Link to="/articles/topic/football">Football</Link>
+          </li>
+          <li>
+            <Link to="/articles/topic/cats">Cats</Link>
```
Comment (pkaodev) [inline src/components/Home.jsx:42]: Unsure why we've purposefully added a link that doesn't show anything

File: src/components/Home.jsx [hunk 3]
```diff
@@ -24,11 +26,22 @@ const Home = () => {
...
+        <h2>Topics</h2>
+        <ul>
+          <li>
+            <Link to="/articles/topic/coding">Coding</Link>
+          </li>
+          <li>
+            <Link to="/articles/topic/cooking">Cooking</Link>
+          </li>
+          <li>
+            <Link to="/articles/topic/football">Football</Link>
```
Comment (pkaodev) [inline src/components/Home.jsx:39]: Right now we've hardcoded in all of the topics, but what happens if someone adds a new topic in the future?

We should be using our APIs GET /api/topics end point to first fetch all available topics, and then use those to dynamically display all available topics

**Comment by pkaodev** [comment]
A couple of things I'm noticing while navigating the page:
-There doesn't seem to be a link to all the articles (I have to enter /articles myself in the url to find this)
-I am not seeing any comments appear.

Could you please change this (and the things I mention in the other comments) and then put another PR in.  Let me know if you want to discuss any of these.

File: src/App.js (hunk 1 above)
Comment (ColemDev) [reply src/App.js:21]: Ah sorry I should have removed my commented-out bits of code.It was mostly there so I wouldn't forget where I should be putting things. 

I guess I'll have to make more use of my notes app so I can keep the planned layout in there instead and leave it out of future production code.

File: src/components/Home.jsx (hunk 2 above)
Comment (ColemDev) [reply src/components/Home.jsx:42]: Doesn't show anything? It should take you to a a list of that topic.

File: src/components/Home.jsx (hunk 3 above)
Comment (ColemDev) [reply src/components/Home.jsx:43]: Okay there is now a dynamic TopicsList component that populates the Navbar component that I've just built with the current  topics.

**Pull request merged**